test/unit/mri.controller.spec.js

```diff
@@ -0,0 +1,320 @@
+            };
+            await mriController.validatorPost(req, res, () => {});
+            assert.strictEqual(reqSpy.callCount, 1);
+            assert.strictEqual(urlSpy.callCount, 1);
+            assert.strictEqual(resSpy.callCount, 0);
+            sinon.restore();
+        });    
+
+        it('should throw errors if validation fails', async function () {
+            let reqSpy = sinon.spy();
+            let urlSpy = sinon.spy();
+            let req = {
+                body: {
+                    atlasName: 'MyAtlas',
+                    atlasProject: 'Visualisation@',
+                    atlasLabelSet: 'SampleLabelSet',
+                    token: 'jnqpincpienfcpewnfcpewn123'
+                },
+                query: {},
+                params: {},
+                value: 0,
+                validationErrors: function() {
+                    return this.body.url ? null : new Error('Invalid url!');
+                },
+                checkBody: sinon.stub().returns({ notEmpty: reqSpy, isURL: urlSpy })
+            };
+            let resSpy = sinon.spy();
+    
+            let res = {
+                status: sinon.stub().returns({ send: sinon.stub().returns({ end: resSpy })}),
+            };
+            await mriController.validatorPost(req, res, () => {});
+            assert.strictEqual(reqSpy.callCount, 1);
+            assert.strictEqual(urlSpy.callCount, 1);
+            assert.strictEqual(resSpy.callCount, 1);
+            sinon.restore();
+        });
+    });
+
+    describe('MRI function() ', function() {
+        it('should return the MRI information when correct input is given', async function() {
+            let req = {
+                db: db,
+                query: {
+                    url: 'https://brainbox.pasteur.fr/mri?url=https://s3.amazonaws.com/fcp-indi/data/Projects/ABIDE_Initiative/Outputs/freesurfer/5.1/CMU_a_0050642/mri/T1.mgz'
+                },
+                dirname: __dirname.split('/test')[0],
+                headers: {},
+                user: {
+                    username: ''
+                },
+                session: {
+                    returnTo: ''
+                },
+                originalUrl: '',
+                isTokenAuthenticated: true,
+                tokenUsername: '',
+                isAuthenticated: function() {
+                    return this.isTokenAuthenticated;
+                },
+                connection: {
+                    remoteAddress: 'http://localhost:3000'
+                }
+            };
+            let arr = [];
+            let authenticated = sinon.stub(req, 'isAuthenticated').resolves(true);
+            let res = {
+                render: sinon.spy()
+            };
+            await mriController.mri(req, res);
+            assert.strictEqual(res.render.callCount, 1);
+            assert.strictEqual(authenticated.callCount, 2);
+            sinon.restore();
+        });
+    });
```

Comment:
does the test do what is in the description? what does mri controller in this test is to look for the provided url in the db, but it doesn't find it so it calls the render function with a generic thing from the request.
in the test we just check that some functions are called a specified number of times, but is it really useful to check that `authenticated` was called twice, given it could be coded differently?

Reply:
@ntraut 
Yup I agree with you on this.
What changes would you suggest?

Reply:
we can try to make in sort that we give a url which is really in the db and look at the json which is given as argument to the render function. i'm not sure how it would be difficult to implement this though

Reply:
Yup @ntraut !
I agree on this.
I'd implement this today and get back to you today.